pyanaconda/core/string.py

```diff
@@ -0,0 +1,176 @@
+#
+# string.py - generic string utility functions
+#
+# Copyright (C) 2021  Red Hat, Inc.  All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+import string  # pylint: disable=deprecated-module
+import sys
+import unicodedata
+
+
+def strip_accents(s):
+    """This function takes arbitrary unicode string
+    and returns it with all the diacritics removed.
+
+    :param s: arbitrary string
+    :type s: str
+
+    :return: s with diacritics removed
+    :rtype: str
+
+    """
+    return ''.join(c for c in unicodedata.normalize('NFD', s)
+                   if unicodedata.category(c) != 'Mn')
+
+
+def ensure_str(str_or_bytes, keep_none=True):
+    """
+    Returns a str instance for given string or ``None`` if requested to keep it.
```

Comment:
Please fix also the doc strings when you are doing these changes.

Reply:
I can't tell what was wrong with this one. But I will change the code anyway, so it's not important.